### PR TITLE
Functionality - Block/Unblock Members

### PR DESCRIPTION
**Pull request by Prashant-singla**
Added a block.js file under modules directory.

### CODE REVIEW

File: modules/block.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+const { MessageType } = require("@adiwajshing/baileys");
+
+module.exports = {
+  name: "block",
```
Comment (Prince-Mendiratta) [inline modules/block.js:4]: Can we please have indentation as 4 spaces.

File: modules/block.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  description: "block contact",
+  extendedDescription:
+    "Send '.block' in a group chat as a reply or in personal chat to block that particular contact.",
+  async handle(client, chat, BotsApp, args) {
+    var sender = chat.messages.all()[0].key.remoteJid;
```
Comment (Prince-Mendiratta) [inline modules/block.js:9]: This is actually defined as a method in BotsApp class, `BotsApp.from`. We can use that here.

File: modules/block.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    var JID = sender.endsWith("@g.us") ?
+      chat.messages.all()[0].message.extendedTextMessage.contextInfo
```
Comment (Prince-Mendiratta) [inline modules/block.js:11]: The linebreaks are weird.

File: modules/block.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+const { MessageType } = require("@adiwajshing/baileys");
+
+module.exports = {
+  name: "block",
+  description: "block contact",
+  extendedDescription:
+    "Send '.block' in a group chat as a reply or in personal chat to block that particular contact.",
+  async handle(client, chat, BotsApp, args) {
+    var sender = chat.messages.all()[0].key.remoteJid;
+    var JID = sender.endsWith("@g.us") ?
```
Comment (Prince-Mendiratta) [inline modules/block.js:10]: Can instead use the class method, `BotsApp.isGroup`.

File: modules/block.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+      chat.messages.all()[0].message.extendedTextMessage.contextInfo
+        .participant : sender;
+    if (args.length === 2) {
+      jidNumber = args[1];
```
Comment (Prince-Mendiratta) [inline modules/block.js:14]: Can you add checks here to make sure the argument is indeed a number and not anything else.

File: modules/block.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      JID = jidNumber + "@s.whatsapp.net";
+    }
+    client.blockUser(JID, args[0] ? args[0] : "add");
+    client.sendMessage(BotsApp.from, "Blocklist Updated", MessageType.text);
```
Comment (Prince-Mendiratta) [inline modules/block.js:18]: You can't really send a message to a blocked person 😛. Can you add a check here to send a message explicitly if the chat is a group.

File: modules/block.js [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+        var blockType = "add";
+
+        if (numberOfArguments > 2) {
+            client.sendMessage(
+                BotsApp.from,
+                "*oops - Number of Arguments exceeded - Try Again :(*",
+                MessageType.text
+            );
+            return;
+        }
```
Comment (Prince-Mendiratta) [inline modules/block.js:20]: Actually there's no need to restrict to strictly 2 args, you can instead check if the first argument can be used and if not, send an error.

File: modules/block.js [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+
+        if (numberOfArguments >= 1) {
+            if (args[0] === "add" || args[0] === "remove") {
+                blockType = args[0];
```
Comment (Prince-Mendiratta) [inline modules/block.js:24]: I didn't quite understand what you're trying to do here?

File: modules/block.js [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+        }
+
+        // Case 1- 2 arguments
+        if (numberOfArguments === 2) {
+            if (args[1].match(/^\d{12}$/)) {
+                var jidNumber = args[1];
+            } else {
+                client.sendMessage(
+                    BotsApp.from,
+                    "*oops - Invalid Contact number - Try Again :(*",
```
Comment (Prince-Mendiratta) [inline modules/block.js:42]: You can send an example here of the type of input you're expecting.

File: modules/block.js [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+        else {
+            if (BotsApp.isGroup) {
+                if (BotsApp.isReply) {
+                    JID =
+                        chat.messages.all()[0].message.extendedTextMessage
+                            .contextInfo.participant;
+                } else {
+                    client.sendMessage(
+                        BotsApp.from,
+                        "*oops - Invalid usecase - Try again :(*",
```
Comment (Prince-Mendiratta) [inline modules/block.js:58]: Better to tell exactly what went wrong so user can see how to fix it.

File: modules/block.js (hunk 8 above)
Comment (Prashant-singla) [reply modules/block.js:24]: It will give error if first argument is anything apart from "add" or "remove"

Comment (Prince-Mendiratta) [reply modules/block.js:16]: Ah, alright.

**Comment by Prince-Mendiratta** [comment]
Thank you!

**Pull request merged**